### ISSUE TITLE
generator raised StopIteration (PEP 479)

### ISSUE THREAD

**Issue by mdickinson** [PyTables/PyTables#784]
We're seeing a `generator raised StopIteration` error in another package that seems to come from the `proxydict` implementation in PyTables.

Example (Python 3.8.1, macOS 10.14.6, PyTables 3.6.1):

```
Python 3.8.1 (default, Dec 22 2019, 03:45:23) 
[Clang 10.0.1 (clang-1001.0.46.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from tables import File
>>> h5file = File("new.h5", "w")
>>> list(h5file.root._v_groups.itervalues())
Traceback (most recent call last):
  File "/Users/mdickinson/.venvs/tables-stop-iteration/lib/python3.8/site-packages/tables/misc/proxydict.py", line 62, in itervalues
    raise StopIteration
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: generator raised StopIteration
```

This is likely related to [PEP 479](https://www.python.org/dev/peps/pep-0479/). The cause seems to be here: https://github.com/PyTables/PyTables/blob/1e7b14e87507c2392265321fe18b2f1f5920ea7f/tables/misc/proxydict.py#L62

I believe that the fix should be as simple as removing the `raise StopIteration` lines in the `iteritems` and `itervalues` methods (along with adding regression tests, of course).

**Comment by avalentino** [comment]
Thanks for reporting @mdickinson.
It should be fixed now (a575bc3f70e0c822eddc3f365ee4ccc0204b9bee)

**Comment by mdickinson** [comment]
@avalentino Thank you for the speedy fix!
